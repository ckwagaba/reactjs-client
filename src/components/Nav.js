import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import store from '../store/Store';
import * as ActionTypes from '../actions/ActionTypes'

class Nav extends Component {
  // handle logout
  handleLogout = (event) => {
    // reset ACCESSTOKEN and userName in Redux store
    store.dispatch({
      type: ActionTypes.STORE_ACCESSTOKEN,
      payload: {
        ACCESSTOKEN: '',
        userName: '',
        userHasAuthenticated: false
      }
    });
    this.props.history.push('/auth/login');
  }

  render () {
    return (
      <nav>
        <div className="nav_links">
          <ul className="menu">
            <li><Link to="/bucketlists/">Bucketlists</Link></li>
          </ul>
        </div>
        <div className="nav_options">
          <ul className="menu">
            <li className="username">{store.getState().authView.userName}</li>
            <li onClick={this.handleLogout} className="logout_button">Logout</li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default withRouter(Nav);
