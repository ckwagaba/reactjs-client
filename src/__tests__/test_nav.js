import React from 'react';
import { shallow } from 'enzyme';
import Nav from '../components/nav';

describe('Nav component', () => {

  const component = <Nav />;

  it('should exist', () => {
    expect(component).toMatchSnapshot();
  });

});
