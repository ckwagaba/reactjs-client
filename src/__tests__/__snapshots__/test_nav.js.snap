// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component should exist 1`] = `<withRouter(Nav) />`;
